x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlDailyMaintenanceService.java

```diff
@@ -70,9 +72,14 @@ public MlDailyMaintenanceService(ClusterName clusterName, ThreadPool threadPool,
     private static TimeValue delayToNextTime(ClusterName clusterName) {
         Random random = new Random(clusterName.hashCode());
         int minutesOffset = random.ints(0, MAX_TIME_OFFSET_MINUTES).findFirst().getAsInt();
-        DateTime now = DateTime.now(ISOChronology.getInstance());
-        DateTime next = now.plusDays(1).withTimeAtStartOfDay().plusMinutes(30).plusMinutes(minutesOffset);
-        return TimeValue.timeValueMillis(next.getMillis() - now.getMillis());
+
+        ZonedDateTime now = ZonedDateTime.now();
+        ZonedDateTime next = now.plusDays(1)
+            .toLocalDate()
+            .atStartOfDay(now.getZone())
+            .plusMinutes(30)
+            .plusMinutes(minutesOffset);
+        return TimeValue.timeValueMillis(next.toInstant().toEpochMilli() - now.toInstant().toEpochMilli());
```

Comment:
we did discuss to use just Instant.now(), but here I think it is better to use ZonedDateTIme since it allows to use convenient DSL for calculating `next` variable

Reply:
+1